pageserver/src/aux_file.rs

```diff
@@ -0,0 +1,100 @@
+use pageserver_api::key::{Key, AUX_KEY_PREFIX, METADATA_KEY_SIZE};
+use tracing::warn;
+
+/// Create a metadata key from a hash, encoded as [AUX_KEY_PREFIX, 2B directory prefix, first 13B of 128b xxhash].
+fn aux_hash_to_metadata_key(prefix: u16, data: &[u8]) -> [u8; METADATA_KEY_SIZE] {
+    let mut key = [0; METADATA_KEY_SIZE];
+    let hash = twox_hash::xxh3::hash128(data).to_be_bytes();
+    let prefix = prefix.to_be_bytes();
```

Comment:
It would be a bit more readable (and perhaps save a few cycles?) to pass in two u8s for the prefix components, rather than writing them as a u16 in encode_aux_file_key and then immediately unpacking them here.

Reply:
fixed in https://github.com/neondatabase/neon/pull/7401/commits/6fc849b7442602b5fbd061bb48d34a624889b21d